app/controllers/tasks_controller.rb

```diff
@@ -3,28 +3,44 @@ class TasksController < ApplicationController
     rescue_from ::ActiveRecord::RecordNotFound, with: :record_not_found
 
     def create
-        task = Task.new(task_params)
-        if task.save
-            render_response(task)
+        if current_user
```

Comment:
All of these actions check to see if a user is logged in, which is resulting in a bunch of repeating/redundant logic. Can we take advantage of using filters here? https://guides.rubyonrails.org/action_controller_overview.html#filters

Reply:
I was thinking to use a filter and return a general message. But then I decides to return a specific message for each method like: "You are not authorized to **create/update/delete** this task."